src/installer/managed/Microsoft.NET.HostModel/Bundle/Manifest.cs

```diff
@@ -65,7 +66,11 @@ private enum HeaderFlags : ulong
         // identify this bundle. It is choosen to be compatible
         // with path-names so that the AppHost can use it in
         // extraction path.
-        public readonly string BundleID;
+        public string BundleID => GetDeterministicId();
+        private string bundleIdInternal;
+        //Same as Path.GetRandomFileName
+        private const int BundleIdLength = 12;
+        private SHA256 hashAlg = SHA256.Create();
```

Comment:
Nit: I would call this something like `bundleHash` or similar - the `hashAlg` reads to me as if it's the algorithm (SHA256 or something else), not the "hash value so far".

Reply:
good point